src/main/java/com/mwt/misc/ChosenAction.java

```diff
@@ -0,0 +1,30 @@
+package com.mwt.misc;
+
+/**
+ * A ChosenAction consists of an action to take,
+ * the probability it was chosen,
+ * and a flag indicating whether to record this decision
+ */
+public class ChosenAction {
```

Comment:
please rename ChosenAction to either DecisionTuple or ExploreResult. I currently use the latter in C#, but will likely switch to the former.